src/components/Popup/Popup.tsx

```diff
@@ -130,12 +142,36 @@ export default class Popup extends AutoControlledComponent<ReactProps<PopupProps
     accessibility: popupBehavior,
     align: 'start',
     position: 'above',
+    on: 'click',
+    mouseLeaveDelay: 500,
   }
 
   public static autoControlledProps = ['open', 'target']
 
   private static isBrowserContext = isBrowser()
 
+  private isPopupClosing = false
+
+  setPopupOpen(newOpen, e) {
+    if (!newOpen) {
+      this.schedulePopupClose(e)
+    } else {
+      this.isPopupClosing = false
+      this.trySetOpen(true, e)
+    }
+  }
+
+  schedulePopupClose = e => {
+    const { mouseLeaveDelay } = this.props
+    this.isPopupClosing = true
+    setTimeout(() => {
+      if (this.isPopupClosing) {
+        this.trySetOpen(false, e)
+      }
+
+      this.isPopupClosing = false
```

Comment:
nit: to move this assignment under `if` condition. 

```js
 if (this.isPopupClosing) {
         this.trySetOpen(false, e)
        this.isPopupClosing = false
   } 
```
So if `this.isPopupClosing` === `false`, you don't need to re-assign it again to `false`

Reply:
Addressed, thanks!